include/alpaka/block/shared/st/BlockSharedMemStMemberMasterSync.hpp

```diff
@@ -0,0 +1,95 @@
+/* Copyright 2019 Benjamin Worpitz, Erik Zenker, Matthias Werner, René Widera
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/block/shared/st/Traits.hpp>
+#include <alpaka/block/shared/st/detail/BlockSharedMemStMemberImpl.hpp>
+#include <alpaka/core/AlignedAlloc.hpp>
+#include <alpaka/core/Common.hpp>
+#include <alpaka/core/Vectorize.hpp>
+
+#include <functional>
+#include <memory>
+#include <vector>
+
+
+namespace alpaka
+{
+    template<std::size_t TDataAlignBytes = core::vectorization::defaultAlignment>
+    class BlockSharedMemStMemberMasterSync
+        : public detail::BlockSharedMemStMemberImpl<TDataAlignBytes>
+        , public concepts::Implements<ConceptBlockSharedSt, BlockSharedMemStMemberMasterSync<TDataAlignBytes>>
+    {
+    public:
+        BlockSharedMemStMemberMasterSync(
+            uint8_t* mem,
+            std::size_t capacity,
+            std::function<void()> fnSync,
+            std::function<bool()> fnIsMasterThread)
+            : detail::BlockSharedMemStMemberImpl<TDataAlignBytes>(mem, capacity)
+            , m_syncFn(fnSync)
+            , m_isMasterThreadFn(fnIsMasterThread)
+        {
+        }
+
+        std::function<void()> m_syncFn;
+        std::function<bool()> m_isMasterThreadFn;
```

Comment:
Are you sure your want to use `std::function`, i.e. a function pointer, here? This could also be solved using a template predicate.

Reply:
It is equal to the implementation before. It based on the synchronized allocation method we have already in alpaka.